web/api/queries/query.c

```diff
@@ -826,12 +861,16 @@ static int rrdr_convert_before_after_to_absolute(
     if(absolute_period_requested == -1)
         absolute_period_requested = 1;
 
-    // make sure they are within our timeframe
-    if(before_requested > last_entry_t)  before_requested = last_entry_t;
+    // Make sure they are within our timeframe
+    // Timeframe can be forward than the "last_entry_t",
+    // due to invalid charts like removed USB disk or stopped application services.
+    // Therefore "last_entry_t" checks/conditions are commented.
+
+    //if(before_requested > last_entry_t)  before_requested = last_entry_t;
```

Comment:
Why are you keep commented the code you are removing?

Reply:
It is closely related with the logic that I changed and would be useful for better understanding. So personally I would like to keep them until next release. But if it is not the right approach from Netdata coding convention perspective then I can sure remove them.